src/rgw/rgw_rest_s3.cc

```diff
@@ -1714,7 +1714,7 @@ int RGWPostObj_ObjStore_S3::get_policy()
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
       S3AuthFactory aplfact(store, s->account_name);
-      RGWGetPolicyLDAPTokenExtractor token_extr(s3_access_key);
+      RGWGetPolicyTokenExtractor token_extr(s3_access_key, received_signature_str);
```

Comment:
I still have some doubts regarding the naming. What is *GetPolicyToken*? Does LDAP have the token concept? If not, I would prefer to not use the "token" here.